File: server/server/sockets/log_name_event.py
import enum


# Use non-standard enum names to match argos enum
# pylint: disable=invalid-name
class LogNameEvent(enum.Enum):
    DroneIds = 'drone-ids'
    BatteryLevel = 'battery-level'
    Orientation = 'orientation'
    Position = 'position'
    Velocity = 'velocity'
    Range = 'range'
    Rssi = 'rssi'
    DroneStatus = 'drone-status'
    Console = 'console'

File: argos/utils/log_name_event.cpp
#include <unordered_map>
#include "log_name_event.h"

namespace {
    std::unordered_map<LogNameEvent, std::string> socketEventStrings = {
        {LogNameEvent::DroneIds, "drone-ids"},
        {LogNameEvent::BatteryLevel, "battery-level"},
        {LogNameEvent::Orientation, "orientation"},
        {LogNameEvent::Position, "position"},
        {LogNameEvent::Velocity, "velocity"},
        {LogNameEvent::Range, "range"},
        {LogNameEvent::Rssi, "rssi"},
        {LogNameEvent::DroneStatus, "drone-status"},
        {LogNameEvent::Console, "console"},
    };
} // namespace

const std::string& logEventToString(const LogNameEvent& event) {
    auto it = socketEventStrings.find(event);
    if (it != socketEventStrings.end()) {
        return it->second;
    }
    return "Unknown";
}
